src/js/common/validator.js

```diff
@@ -572,4 +592,28 @@ var fluid  = fluid  || {};
         });
         return requiredChildren;
     };
+
+    // A global component that validates material using the above static functions.  Handles the expensive initial
+    // compilation of the schema.  For performance reasons, this is not itself a schema-validated component.
+    fluid.defaults("gpii.schema.validator", {
+        gradeNames: ["fluid.component", "fluid.resolveRootSingle"],
+        singleRootType: "gpii.schema.validator",
+        members: {
+            validatorsByHash: {
+            }
+        },
+        invokers: {
+            validate: {
+                funcName: "gpii.schema.validator.validate",
+                args:     ["{that}", "{arguments}.0", "{arguments}.1", "{arguments}.2"] // gssSchema, toValidate, schemaHash
+            },
+            clearCache: {
+                funcName: "fluid.set",
```

Comment:
Whilst clearing the whole cache is probably a worthwhile operation to support, it is more likely that the user will want the more controllable option of evicting a the cache for a particular schema that they know they will not be using again. The operation should accept the gssSchema which is to be evicted, and call a "delete" on the matching cache slot.

Reply:
I went just a little further, as I would like to have a place to sketch more performant alternatives off to the side.  I added an invoker to remove a single schema from the cache, and also one to cache a single schema.  I also restored the ability to pass in a schema hash throughout.  This leaves the option to further optimise in derived grades without requiring us to do the work now.